### PR TITLE
General compat refactor

### PR DESCRIPTION
**Pull request by vimwitch**
Pulls the unirep-social specific parts of the synchronizer into this package.

Compatibility updates with unirep monorepo.

### CODE REVIEW

File: scripts/deploy.js [hunk 1]
```diff
@@ -22,9 +22,9 @@ const maxAttesters = 2 ** circuitUserStateTreeDepth - 1
...
         {
-            globalStateTreeDepth: 5,
-            userStateTreeDepth: 5,
-            epochTreeDepth: 32,
+            globalStateTreeDepth,
+            userStateTreeDepth,
+            epochTreeDepth,
         },
         {
             attestingFee,
```
Comment (vivianjeng) [inline scripts/deploy.js:30]: actually they are in the same config
`const deployUnirep = async (deployer, _settings) => {}`
so set the config together if you use `deployUnirep` 😅
e.g.
```typescript
{
    globalStateTreeDepth,
    userStateTreeDepth,
    epochTreeDepth,
    attestingFee,
    ...
}
```

File: package.json [hunk 2]
```diff
@@ -23,32 +23,33 @@
     },
     "homepage": "https://github.com/Unirep/Unirep-Social#readme",
     "devDependencies": {
+        "@nomicfoundation/hardhat-chai-matchers": "^1.0.0-beta.2",
         "@nomiclabs/hardhat-ethers": "^2.0.2",
-        "@nomiclabs/hardhat-waffle": "^2.0.1",
```
[the remark this answers is not in the document]
Comment (vimwitch) [reply package.json:27]: Possibly, I switched to [@nomicfoundation/hardhat-chai-matchers](https://www.npmjs.com/package/@nomicfoundation/hardhat-chai-matchers) so I could use `revertedWithCustomError` to check our custom error types. I think it's a good idea to use it in our other packages, but no rush.

File: core/synchronizer.ts [hunk 3]
```diff
@@ -0,0 +1,805 @@
...
+        const existingNullifier = await this._db.findOne('Nullifier', {
+            where: {
+                nullifier: repNullifiers,
+                confirmed: true,
+            },
+        })
+        if (existingNullifier) {
+            console.log(`comment duplicated nullifier`, repNullifiers)
+            return
+        }
```
[the remark this answers is not in the document]
Comment (vimwitch) [reply core/synchronizer.ts:258]: Gotcha, I'm going to keep it here until we add it in the unirep synchronizer. I'd rather check it twice than not at all.

**Comment by vimwitch** [comment]
@vivianjeng this should be ready, all tests passing

File: core/synchronizer.ts [hunk 4]
```diff
@@ -0,0 +1,805 @@
...
+        const _epoch = Number(event.topics[1])
+        const _commitment = BigInt(event.topics[2]).toString()
+        db.create('UserSignUp', {
+            transactionHash: event.transactionHash,
+            commitment: _commitment,
+            epoch: _epoch,
+        })
+    }
+
+    private async verifyAttestationProof(
```
Comment (vivianjeng) [inline core/synchronizer.ts:127]: If it is verified by `@unirep/core` Synchronizer, we can remove the check here.

Comment (vimwitch) [reply core/synchronizer.ts:127]: This is fixed, but it needs unirep/unirep#58 to work properly

Comment (vimwitch) [reply core/synchronizer.ts:127]: Moved to #11 so we can merge this

**Pull request merged**